war/src/main/js/add-item.js

```diff
@@ -125,22 +123,93 @@ $.when(getItems()).done(function(data){
     //////////////////////////
     // helper functions...
 
+    function setSelectState(e){
+      if(e.preventDefault){
+        e.preventDefault();
+      }
+      else{
+        e = {currentTarget:e};
+      }
+      var href = $(e.target).attr('href');
+      if(href) {
+        window.open(href);
+      }
+      
+      var $this = $(e.currentTarget).closest('li');
+      var $cat = $this.closest('.categories');
+      var $radios = $cat.find('input[type="radio"][name="mode"]').removeProp('checked');
+      var $checked = $radios.find(':checked');
+      
+      $cat.find('.active').removeClass('active');
+      
+      // Something in Jenkins land wants to reverse the radio button change on keydown. Waiting a pulse avoids the issue.
+      setTimeout(function(){
+        $this.addClass('active');
+        $this.find('input[type="radio"]').prop('checked', true);
+        checkFormReady();
+      },1);       
+    }    
+    
     function checkFormReady(){
+      
       //make sure everyone has changed and gotten attached...
       setTimeout(function(){
         var $name = $form.removeClass('no-select').find('input[name="name"]').removeClass('no-val');
+        var $nameBox = $name.closest('.j-add-item-name');
+        var val = $.trim($name.val());
+        var named = false;
+        
+        $nameBox.parent().find('.error').remove();
         
-        function checkItems(){
-          var selected = $form.find('input[type="radio"]:checked').length > 0;
-          var named = $.trim($name.val()).length > 0;
-          return {selected:selected,named:named};
+        function checkJobName(){
+          if(val.length === 0 ){
+            checkName('<div class="error">You must name this item.</div>');
+            return confirmBoth(false);
+          }
+          $.get('../../checkJobName?value='+val).always(checkName);
```

Comment:
Looking at the 1.x code, which was wrong—always called `checkUrl="${rootURL}/checkJobName"` ~ `Jenkins.doCheckJobName`, which I find is tracked as [JENKINS-20509](https://issues.jenkins-ci.org/browse/JENKINS-20509). Would like to fix it, but will require backend changes as well as frontend, because we need to move `doCheckJobName` to `ViewDescriptor`. (Which is compatible for existing calls at root level, but to benefit from the fix, calls from a folder then need to stop using the absolute path `/checkJobName` and start using simply `checkJobName`—on the same level as `/job/folder/view/All/newJob`, so would hit `/job/folder/view/All/checkJobName`.) I will work on it.